aptos-move/block-executor/src/executor.rs

```diff
@@ -479,25 +514,13 @@ where
                     }
                     // layout is Some(_) if it contains an delayed field
                     if let DataRead::Versioned(_version, value, Some(layout)) = data_read {
-                        if let Some(value_bytes) = value.bytes() {
-                            match latest_view.replace_identifiers_with_values(value_bytes, layout) {
-                                Ok((patched_bytes, delayed_field_keys_in_resource)) => {
-                                    if !delayed_field_keys
-                                        .is_disjoint(&delayed_field_keys_in_resource)
-                                    {
-                                        let mut patched_value = value.as_ref().clone();
-                                        patched_value.set_bytes(patched_bytes);
-                                        patched_resource_write_set
-                                            .insert(key.clone(), patched_value);
-                                    }
-                                },
-                                Err(_) => unreachable!(
-                                    "Failed to replace identifiers with values in read set"
-                                ),
-                            };
-                        } else {
-                            // TODO: Is this unreachable?
-                            unreachable!("Data read value must exist");
+                        if let Some(patched_value) = Self::replace_ids_with_values(
+                            value.clone(),
```

Comment:
nit: you can `as_ref()` to pass a reference instead of cloning the arc

Reply:
Tried it. I am using a clone operation inside replace_ids_with_values. So, had to use Arc pointer instead of references.